packages/devtools_app/lib/src/survey.dart

```diff
@@ -29,11 +29,11 @@ class SurveyService {
 
   Future<DevToolsSurvey> get activeSurvey async {
     _cachedSurvey ??= await _fetchSurveyContent();
-    if (_cachedSurvey != null) {
+    if (_cachedSurvey != null && server.isDevToolsServerAvailable) {
```

Comment:
we should add this check before we fetch the survey content and return null if false, because if the server is unavailable we don't need to do anything survey related

Reply:
that is better. done.